src/navigation-plan.js

```diff
@@ -71,17 +81,13 @@ export function buildNavigationPlan(navigationContext, forceLifecycleMinimum) {
 }
 
 export class BuildNavigationPlanStep {
-	run(navigationContext, next) {
-    if (navigationContext.nextInstruction.config.redirect) {
-      return next.cancel(new Redirect(navigationContext.nextInstruction.config.redirect));
-    }
-
+  run(navigationContext, next) {
     return buildNavigationPlan(navigationContext)
       .then(plan => {
         navigationContext.plan = plan;
         return next();
       }).catch(next.cancel);
```

Comment:
This `catch(next.cancel)` will now trigger the redirect previously handled above.